lxd/db/cluster/config.mapper.go

```diff
@@ -39,12 +39,36 @@ func GetConfig(ctx context.Context, tx *sql.Tx, parent string) (map[int]map[stri
 		fillParent[i] = strings.Replace(parent, "_", "s_", -1) + "s"
 	}
 
-	sqlStmt, err := prepare(tx, fmt.Sprintf(configObjectsLocal, fillParent...))
-	if err != nil {
-		return nil, err
+	queryStr := fmt.Sprintf(configObjectsLocal, fillParent...)
+	queryParts := strings.SplitN(queryStr, "ORDER BY", 2)
+	args := make([]any, 0, DqliteMaxParams)
+
```

Comment:
This function seems to be missing a check that generates an error if a filter is provided that has both key and value set to nil (the other functions seem to have a check for it though).

Reply:
Not sure what you mean here? If both key and value are set to nil, that's an empty filter, which is valid. 

The WHERE clause will only be appended to the query if at least one of the fields is non-nil.

Though now I've noticed that I accidentally kept the slice fields for the `Config` and `Devices` filters. I'll fix that now.

Reply:
> If both key and value are set to nil, that's an empty filter, which is valid.

I dont think we should allow passing in an empty filter struct. It doesn't make any sense to me.

Reply:
I suppose now that the filters are variadic, it changes things a bit, but before this point you had to include a filter, so about half of all of the filters we're using right now are in fact empty.

The other functions also never errored out in that case. It was always the final valid case. Instances for example:
```go
} else if filter.ID == nil && filter.Project == nil && filter.Name == nil && filter.Node == nil && filter.Type == nil {   
 sqlStmt = Stmt(tx, instanceObjects)
} else {
 return nil, fmt.Errorf("No statement exists for the given Filter")
}
```

Reply:
Yeah it does change things a bit, but I think it would make it less accident prone to prevent the use of empty filter structs.